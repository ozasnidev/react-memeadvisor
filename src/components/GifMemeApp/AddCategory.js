import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [newCategory, setNewCategory] = useState('');
    const handleNewInputValue = (event) => {
        setNewCategory(event.target.value);
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        if(newCategory.trim().length > 2){
            setCategories(categories => [newCategory, ...categories]);
            setNewCategory('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="panel-block">
            <p className="control">
                <input className="input is-success" 
                    type="text" 
                    value={newCategory}
                    onChange={handleNewInputValue}
                    placeholder="Search" 
                />
            </p>
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
