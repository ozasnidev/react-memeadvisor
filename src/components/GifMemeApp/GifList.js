import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import PropTypes from 'prop-types';

export const GifList = ({category}) => {

    const {data,loading} = useFetchGif( category );

    return (
        <div>
            <div className="panel-block">
                <h4 className="title is-4 animate__animated animate__fadeInDown"> {category} </h4>
                {loading && <div className="spinner"></div>}
            </div>
            <div className="img-thumbnail-container">
                {data.map(image => ( 
                    <img src={image.url} alt={image.title} key={image.id} className="img-thumbnail  animate__animated animate__fadeInUp"/>
                ))}
            </div>
        </div>
    )
}

GifList.propTypes = {
    category: PropTypes.string.isRequired
}