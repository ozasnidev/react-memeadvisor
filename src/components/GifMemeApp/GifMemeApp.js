import React, {useState} from 'react';
import NavbarMeme from '../Shared/navbar/Navbar';
import { AddCategory } from './AddCategory';
import { GifList } from './GifList';


const GifMemeApp = ({ defaultCategories = []}) => { 
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <NavbarMeme />
            <div className="container">
                <div className="add-category-container">
                    <article className="panel is-success">
                        <p className="nav-title panel-heading">Categories</p>
                        <AddCategory setCategories = {setCategories}/>
                        {categories.map(category => <GifList key={category} category={category} />)} 
                    </article>
                </div>
            </div>
        </>
    )   
}
export default GifMemeApp;