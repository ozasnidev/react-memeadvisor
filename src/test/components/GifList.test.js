import { shallow } from 'enzyme';
import React from 'react';
import { GifList } from '../../components/GifMemeApp/GifList';

describe('GifList component test', () => {
    const categoryTest = 'goku';
    const wrapper = shallow(<GifList category={categoryTest}/>);
    
    test('should to show GifList component without issues', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should to have a h4 tag with a right title', () => {
        const h4Tag = wrapper.find('h4');
        expect(h4Tag.text().trim()).toBe(categoryTest);
    });
    test('should to contain <div className="img-thumbnail-container">', () => {
        expect(wrapper.find('div')
            .at(3)
            .hasClass('img-thumbnail-container'))
                .toBeTruthy();
    });
});