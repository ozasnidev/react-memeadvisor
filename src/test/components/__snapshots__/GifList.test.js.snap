// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifList component test should to show GifList component without issues 1`] = `
<div>
  <div
    className="panel-block"
  >
    <h4
      className="title is-4 animate__animated animate__fadeInDown"
    >
       
      goku
       
    </h4>
    <div
      className="spinner"
    />
  </div>
  <div
    className="img-thumbnail-container"
  />
</div>
`;

exports[`GifList component test should to show images using useFetchGif hook 1`] = `
<div>
  <div
    className="panel-block"
  >
    <h4
      className="title is-4 animate__animated animate__fadeInDown"
    >
       
      goku
       
    </h4>
  </div>
  <div
    className="img-thumbnail-container"
  >
    <img
      alt="something"
      className="img-thumbnail  animate__animated animate__fadeInUp"
      key="1234"
      src="https://localhost/something.com"
    />
  </div>
</div>
`;
